### ISSUE TITLE
your-username@your-hostname

### ISSUE THREAD

**Issue by J0G** [Misterio77/nix-starter-configs#52]
Hi Misterio77,

I am learning a great deal with your starter configs. Thank you :)

There is one thing though I struggle to make sense of when using the standalone version of home-manager:

I wonder what is the benefit from using:
`homeConfigurations = {
   "your-username@your-hostname" = home-manager.lib.homeManagerConfiguration {`

Intead of : 
`homeConfigurations = {
   user-name = home-manager.lib.homeManagerConfiguration {`

Is there actually a difference? Maybe it just come down to preference or good practice. 
To my newbie eyes the "your-username@your-hostname" seems redundant, for instance if you have two hosts and the same user. It also prevents you from using the simple `home-manager switch --flake .` shortcut when switching to a new home manager generation.
Or maybe it's a way to tweak your user setting per host.

PS: learning github as well, I do not have all the codes yet.

**Comment by Misterio77** [comment]
> I am learning a great deal with your starter configs. Thank you :)

I'm glad to be of help :D 

> Intead of : homeConfigurations = { user-name = home-manager.lib.homeManagerConfiguration {

I actually didn't even know that I was possible haha. I wonder if it's a recently added (or recently documented) feature.

On my personal config, I do specify the host so that I can have different configs on each of them; but this host-less alternative might be useful for people who have a single host or don't need different configs per-host.

> It also prevents you from using the simple home-manager switch --flake . shortcut when switching to a new home manager generation.

It should(tm) work though (def works here). Using `--flake .` will basically default to "--flake .$(whoami)@$(hostname)".

**Comment by J0G** [comment]
My comment originated from a video I followed on youtube, from LibrePhoenix that made flakes look less mystical.

I actually went with your-username@your-hostname, and after a few days of testing I now see the benefit. My main user doesn't need to be the same on all machines if they serve a different purpose. 

Doing it this way also solved one of my issues I had before where running `home-manager switch-- flake .` and  `home-manager switch --flake.#username` yielded different results, the later one not working. Weird thing.

I still don't get everything that's going on in my flake, but I get enough to make it work and learn as I go. It's kind of a little garden. Thanks for the seed though :)
